packages/main/src/MenuItemGroup.ts

```diff
@@ -0,0 +1,134 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import jsxRenderer from "@ui5/webcomponents-base/dist/renderer/JsxRenderer.js";
+import type MenuItem from "./MenuItem.js";
+import MenuItemGroupTemplate from "./MenuItemGroupTemplate.js";
+import { isInstanceOfMenuItem } from "./MenuItem.js";
+import ItemCheckMode from "./types/ItemCheckMode.js";
+import type { IMenuItem } from "./Menu.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * The `ui5-menu-item-group` component represents a group of items designed for use inside a `ui5-menu`.
+ * Items belonging to the same group should be wrapped by a `ui5-menu-item-group`.
+ * Each group can have an `itemCheckMode` property, which defines the check mode for the items within the group.
+ * The possible values for `itemCheckMode` are:
+ * - 'None' (default) - no items can be checked
+ * - 'Single' - Only one item can be checked at a time
+ * - 'Multiple' - Multiple items can be checked simultaneously
+ *
+ * **Note:** If the `itemCheckMode` property is set to 'Single', only one item can remain checked at any given time.
+ * If multiple items are marked as checked, the last checked item will take precedence.
+ *
+ * ### Usage
+ *
+ * `ui5-menu-item-group` represents a collection of `ui5-menu-item` components that can have the same check mode.
+ * The items are addeed to the group's `items` slot.
+ *
+ * ### ES6 Module Import
+ *
+ * `import "@ui5/webcomponents/dist/MenuItemGroup.js";`
+ * @constructor
+ * @extends UI5Element
+ * @implements {IMenuItem}
+ * @since 2.11.0
+ * @public
+ */
+@customElement({
+	tag: "ui5-menu-item-group",
+	renderer: jsxRenderer,
+	template: MenuItemGroupTemplate,
+})
+class MenuItemGroup extends UI5Element implements IMenuItem {
+	/**
+	 * Defines the component's check mode.
+	 * @default "None"
+	 * @public
+	 */
+	@property()
+	itemCheckMode: `${ItemCheckMode}` = "None";
```

Comment:
I think **item** is obsolete in the prop's name, It's better be **checkMode** (or whatever the right word is checkMode, checkingMode, I guess check sounds better), same as in the  in the List - **selectionMode**, not itemSelectionMode.

Reply:
Done